bin/find-spark-home

```diff
@@ -34,8 +34,5 @@ elif [ ! -f "$FIND_SPARK_HOME_PYTHON_SCRIPT" ]; then
 else
   # We are pip installed, use the Python script to resolve a reasonable SPARK_HOME
   # Default to standard python interpreter unless told otherwise
-  if [[ -z "$PYSPARK_DRIVER_PYTHON" ]]; then
-     PYSPARK_DRIVER_PYTHON="${PYSPARK_PYTHON:-"python"}"
-  fi
-  export SPARK_HOME=$($PYSPARK_DRIVER_PYTHON "$FIND_SPARK_HOME_PYTHON_SCRIPT")
+  export SPARK_HOME=$(${PYSPARK_PYTHON:-"python"} "$FIND_SPARK_HOME_PYTHON_SCRIPT")
```

Comment:
Hmmm .. can we strip the non-printable characters instead?
Respecting `PYSPARK_DRIVER_PYTHON` falling back to `PYSPARK_PYTHON` is expected.

Reply:
At least I can come up with one way although it's hacky. e.g.)

```bash
a=$(ipython -c "import sys; print('/User', file=sys.stderr)" 2>&1 >/dev/null)
ls $a
```

Reply:
That's a workaround for `ipython`, but not for `jupyter`, because `jupyter` doesn't support `jupyter find_spark_home.py`. I think `PYSPARK_DRIVER_PYTHON` is more meant for "frontend". This fix enables `
PYSPARK_DRIVER_PYTHON=jupyter PYSPARK_DRIVER_PYTHON_OPTS=notebook  pyspark` too.

Reply:
Btw,  `PYSPARK_DRIVER_PYTHON`'s falling back to `PYSPARK_PYTHON` happens after `find-spark-home` too: https://github.com/apache/spark/blob/f1fde0cc2272d5c04cf25960be1ebd46001c5354/bin/pyspark#L45

Reply:
@mzhang-code, can we just add a bandaid fix like: if `PYSPARK_DRIVER_PYTHON` ends with `jupyter` or `ipython`, uses `PYSPARK_PYTHON` or `python` for now with some comments about why we're using `PYSPARK_PYTHON` instead of `PYSPARK_DRIVER_PYTHON`?